src/main.go

```diff
@@ -128,11 +138,106 @@ func (app *App) parse(b []byte) ([]byte, error, error) {
 	return pc, errParse, err
 }
 
+func (app *App) parseRemoteURL(urlString string) ([]byte, error, error) {
+	log.Infof("called parseRemoteURL() url: %s", urlString)
+	p := publiccode.NewParser()
+	errParse := p.ParseRemoteFile(urlString)
+	pc, err := p.ToYAML()
+
+	return pc, errParse, err
+}
+
+func promptError(err error, w http.ResponseWriter,
+	httpStatus int, mess string) {
+
+	log.Errorf(mess+": %v", err)
+
+	message := Message{
+		Status:  httpStatus,
+		Message: mess,
+		Error:   err.Error(),
+	}
+	log.Debugf("message: %v", message)
+	w.Header().Set("Content-type", "application/json")
+	w.WriteHeader(message.Status)
+	json.NewEncoder(w).Encode(message)
+}
+
+func promptValidationErrors(err error, w http.ResponseWriter,
+	httpStatus int, mess string) {
+
+	log.Errorf(mess+": %v", err)
+
+	message := Message{
+		Status:          httpStatus,
+		Message:         mess,
+		ValidationError: errorsToSlice(err),
+	}
+
+	w.Header().Set("Content-type", "application/json")
+	w.WriteHeader(message.Status)
+	messageJSON, _ := json.Marshal(message)
+	w.Write(messageJSON)
+}
+
+func errorsToSlice(errs error) (arr []ErrorInvalidValue) {
+	keys := strings.Split(errs.Error(), "\n")
+	for _, key := range keys {
+		arr = append(arr, ErrorInvalidValue{
+			Key: key,
+		})
+	}
+	return
+}
+
+func (app *App) validateRemoteURL(w http.ResponseWriter, r *http.Request) {
+	log.Info("called validateFromURL()")
+	setupResponse(&w, r)
+	if (*r).Method == "OPTIONS" {
+		return
+	}
+	// Getting vars from parameters
+	vars := mux.Vars(r)
+	urlString := vars["url"]
+	if urlString == "" {
+		promptError(errors.New("Not found"), w, http.StatusNotFound, "URL error")
+		return
+	}
+
+	//parsing
+	pc, errParse, errConverting := app.parseRemoteURL(urlString)
+
+	if errConverting != nil {
+		log.Debugf("Error converting: %s", errConverting)
+		promptError(errConverting, w, http.StatusBadRequest, "Error converting")
+	}
+	if errParse != nil {
+		if match, _ := regexp.MatchString(`404`, errParse.Error()); match {
+			log.Debugf("URL Error: %s", errParse)
+			promptError(errors.New("Not found"), w, http.StatusNotFound, "URL error")
+			return
+		}
+		log.Debugf("Validation Errors: %s", errParse)
```

Comment:
Same as above. I'd try to print a debug message inside the error wrapper function, what do you think?